extract_unique_sensors_01.py

```diff
@@ -0,0 +1,28 @@
+from urllib.request import urlopen
+
+sensors_data = urlopen('http://monctl.devm.camlab.kat.ac.za/kat/doc/manuals/sensors/SensorList.txt')
+lines = sensors_data.readlines()
+
+
+def sensors(lines):
+    #global lines
+    candidate_lines = []
+    for line in lines:
+        line = line.decode()
+        if line.startswith('kat.sensors'):
+            for candidate_line in line.split():
+                candidate_lines.append(candidate_line)
+    return candidate_lines
+
+# Loop over candidate_lines and extract sensor names
+def sensor_names():
+    sensors_list = []
+    candidate_lines = sensors(lines)
+    for sensor in candidate_lines:
+    # Typical output from "sensor" variable:
+    # kat.sensors.anc_api_version   kat.sensors.anc_ganglia_api_version   kat.sensors.anc_ganglia_kat_monctl_system_os_name
+        sensor = sensor.split('.')[-1]  
+        sensor = sensor.strip("\n") 
+        sensors_list.append(sensor)
+    return sensors_list 
+print(sensor_names())
```

Comment:
Missing blank line at end of your script!

Ideally the usage needs to be added under `if __name__ == "__main__":`

See: https://stackoverflow.com/questions/419163/what-does-if-name-main-do
and, https://docs.python.org/3.6/library/__main__.html

Reply:
@mamkhari